### PR TITLE
chore: change builtin dashboard owner

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [x] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/reporting/parameter.ts [hunk 1]
```diff
@@ -69,10 +69,18 @@ export function createStackParametersQuickSight(scope: Construct, paramGroups?:
     default: 'Comma Delimited Subnet Ids',
   };
 
-  const quickSightPrincipalParam = new CfnParameter(scope, 'QuickSightPrincipalParam', {
+  const quickSightOwnerPrincipalParam = new CfnParameter(scope, 'QuickSightOwnerPrincipalParam', {
```
Comment (zxkane) [inline src/reporting/parameter.ts:72]: Let's add the pattern to require a valid QuickSight ARN.

File: src/reporting/parameter.ts [hunk 2]
```diff
@@ -69,10 +69,18 @@ export function createStackParametersQuickSight(scope: Construct, paramGroups?:
...
     description: 'Arn of the QuickSight principal, QuickSight resource will be owned by this principal.',
     type: 'String',
   });
+  labels[quickSightOwnerPrincipalParam.logicalId] = {
+    default: 'QuickSight Owner Principal Arn',
+  };
+
+  const quickSightPrincipalParam = new CfnParameter(scope, 'QuickSightPrincipalParam', {
```
Comment (zxkane) [inline src/reporting/parameter.ts:80]: Let's add the pattern to require a valid QuickSight ARN.

File: src/reporting/private/dashboard.ts [hunk 3]
```diff
@@ -91,10 +106,21 @@ export const analysisPermissionActions = [
...
 
-export const dashboardPermissionActions = [
+export const dashboardReaderPermissionActions = [
+  'quicksight:DescribeDashboard',
+  'quicksight:ListDashboardVersions',
+  'quicksight:QueryDashboard',
+];
+
+export const dashboardAdminPermissionActions = [
   'quicksight:DescribeDashboard',
```
Comment (zxkane) [inline src/reporting/private/dashboard.ts:116]: use `...dashboardReaderPermissionActions` to reuse the reader's permission set

File: src/reporting/private/dashboard.ts [hunk 4]
```diff
@@ -73,15 +75,28 @@ export interface QuickSightDashboardDefProps {
...
+export const dataSetReaderPermissionActions = [
   'quicksight:DescribeDataSet',
   'quicksight:DescribeDataSetPermissions',
   'quicksight:PassDataSet',
   'quicksight:DescribeIngestion',
   'quicksight:ListIngestions',
 ];
 
-export const analysisPermissionActions = [
+export const dataSetAdminPermissionActions = [
```
Comment (zxkane) [inline src/reporting/private/dashboard.ts:86]: use `...dataSetReaderPermissionActions` to reuse the reader's permission set

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/reporting/private/dashboard.ts (hunk 3 above)
Comment (llmin) [reply src/reporting/private/dashboard.ts:116]: fixed

File: src/reporting/private/dashboard.ts (hunk 4 above)
Comment (llmin) [reply src/reporting/private/dashboard.ts:86]: fixed

File: src/reporting/parameter.ts (hunk 2 above)
Comment (llmin) [reply src/reporting/parameter.ts:82]: fixed

File: src/reporting/lambda/custom-resource/quicksight/index.ts [hunk 5]
```diff
@@ -465,27 +491,31 @@ const createAnalysis = async (quickSight: QuickSight, awsAccountId: string, prin
...
       Name: `${props.dashboardName} - ${identifier.schemaIdentifier} - ${identifier.databaseIdentifier} `,
       Permissions: [{
-        Principal: principalArn,
-        Actions: dashboardPermissionActions,
+        Principal: commonParams.ownerPrincipalArn,
+        Actions: dashboardAdminPermissionActions,
+      },
+      {
+        Principal: commonParams.sharePrincipalArn,
+        Actions: dashboardAdminPermissionActions,
```
[the remark this answers is not in the document]
Comment (zxkane) [reply src/reporting/lambda/custom-resource/quicksight/index.ts:512]: We need deep dive which dashboard permission to let a user save it as an analysis. It does not make sense to grant the delete dashboard permission to a shared user.

And we change the shared principal to read permission only when updating the dashboard.

https://github.com/awslabs/clickstream-analytics-on-aws/pull/448/files#diff-b9a6f2b05a847550baa6c787085a75f2aa064cdfa16d31f6d5a56bf36f62822cR805-R806

We need to make them consistent.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/reporting/lambda/custom-resource/quicksight/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (llmin) [reply src/reporting/lambda/custom-resource/quicksight/index.ts:512]: These permissions do not support arbitrary combinations; we can only choose two combinations. 
One is read-only, and the other is the current combination.

**Review by tyyzqmf — APPROVED** [review]

**Review by zxkane — APPROVED** [review]

**Pull request merged**